cmd/find.go

```diff
@@ -190,142 +199,129 @@ func doFind(clnt Client, ctx *cli.Context) {
 		}
 
 		fileContent := parseContent(content)
-		filePath := fileContent.Key
-
-		// traversing in a object store not a file path
-		if err == nil {
-			filePath = path.Join(alias, filePath)
+		if targetAlias != "" {
+			fileContent.Key = path.Join(targetAlias, fileContent.Key)
 		}
 
 		if ctx.String("maxdepth") != "" {
-			i, e := strconv.Atoi(ctx.String("maxdepth"))
-			s := ""
-
+			maxDepth, e := strconv.Atoi(ctx.String("maxdepth"))
 			fatalIf(probe.NewError(e), "Error parsing string passed to flag maxdepth")
 
-			// we are going to be parsing the path by x amounts
-			pathParts := strings.SplitAfter(filePath, "/")
+			newPath := ""
 
-			// handle invalid params
-			// ex. user specifies:
-			// maxdepth 2, but the given object only has a maxdepth of 1
-			if (len(pathParts)-1) < i || i < 0 {
+			// We are going to be parsing the path by x amounts
+			pathParts := strings.SplitAfter(fileContent.Key, separator)
 
-				// -1 is meant to handle each the array being 0 indexed, but the size not being 0 indexed
-				i = len(pathParts) - 1
+			// Check if max-depth is 2, but the given object only
```

Comment:
`but the given object only has a maximum depth of 1 use that instead.`
=>
`but if the given object only has a maximum depth of 1, then use that instead.`

Reply:
Done